internal/core/regularsurface.hpp

```diff
@@ -67,6 +88,15 @@ class RegularSurface{
         std::size_t const col
     ) const noexcept (false);
 
+    Point to_cdp(
+        std::size_t i
+    ) const noexcept(false);
+
+    /* World coordinates -> grid position */
+    Point from_cdp(
+        Point point
+    ) const noexcept (false);
+
```

Comment:
And there is one more above which goes from row, col! :smile: 

But it is not that parameters are symmetrical, it's the logic. You can have `Point to_cdp(Point)` as well, and it would be the same. Only that we never  needed this representation.
So I think it's fine.